pkg/KubeArmorController/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/kubearmor/KubeArmor/pkg/KubeArmorController
 
-go 1.21
+go 1.22.1
```

Comment:
I think here I would suggest using only `1.22` otherwise `go` will always try to pull the updated patch version. 
So, if you merge a PR for the patch `1.22.6` then on every system go will try to update it to 1.22.6 even if you run `go version` command in the directory. 

If you use `1.22` then it will support all the 1.22 patch series. 
Again, if you want that everyone should use what we have in patch releases then it's good but I think it's way frequent go download.

Reply:
yes, but it seems 1.22.1 has the vulnerability fixes:
https://pkg.go.dev/vuln/GO-2024-2600
https://pkg.go.dev/vuln/GO-2024-2599
https://pkg.go.dev/vuln/GO-2024-2598

on 1.22 the go-vuln check will always fail

Reply:
But certain vulnerabilities  are being handled in go 1.22.1.

Reply:
Yes, that is true. This is just so that developers have ease with development. 

If we patch go.mod with exact version very frequently then one have to download a new go version on every patch. 

This has nothing to do with production, we are already using patched version as a image and that's important and easy part because we build it using an image which is already patched. 

This point is about developing locally. 
I'm using go 1.22.0 locally and when I checkout your PR and run any go command, I get this.
```bash
$ go version
go: downloading go1.22.1 (linux/amd64)
```
When I update the same to go `1.22` in go.mod, it works fine. 

```diff
$ git diff -U0
diff --git a/KubeArmor/go.mod b/KubeArmor/go.mod
index 36dfd9f8..6d06d1da 100644
--- a/KubeArmor/go.mod
+++ b/KubeArmor/go.mod
@@ -3 +3 @@ module github.com/kubearmor/KubeArmor/KubeArmor
-go 1.22.1
+go 1.22
```
```bash
$ go version
go version go1.22.0 linux/amd64
```
So, do you want that every developer working on the project updates go on his system whenever we patch then please proceed with `1.22.1`
I think for someone this might be too early but given it's for a good and secure reason, I'm okay with this. I want to use the latest go version. :) 
We can proceed with the same.